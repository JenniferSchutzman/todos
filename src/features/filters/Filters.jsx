/* eslint-disable no-shadow */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import {
  Button,
  Form,
  Popup,
} from 'semantic-ui-react';

import { setLabels, setShowCompleted } from './filtersReducer';

const mapState = () => ({});

const mapDispatch = {
  setLabels,
  setShowCompleted,
};

class Filters extends Component {
  state = {
    filters: '',
    open: false,
    showCompleted: false,
  };

  handleClose = () => this.setState({ open: false });

  handleFilters = (event) => {
    const { setLabels } = this.props;

    this.setState(
      { [event.target.name]: event.target.value },
      () => {
        const { filters } = this.state;

        setLabels(filters);
      },
    );
  };

  handleOpen = () => this.setState({ open: true });

  handleShowCompleted = () => {
    const { setShowCompleted } = this.props;

    this.setState(
      (prevState) => ({ showCompleted: !prevState.showCompleted }),
      () => {
        const { showCompleted } = this.state;

        setShowCompleted(showCompleted);
      },
    );
  };

  render() {
    const { filters, open, showCompleted } = this.state;

    return (
      <Popup
        on="click"
        open={open}
        onClose={this.handleClose}
        onOpen={this.handleOpen}
        position="bottom right"
        trigger={<Button color="green" icon="settings" inverted />}
      >
        <Popup.Header>Filters</Popup.Header>
        <Popup.Content>
          <Form>
            <Form.Input
              id="todo-filter_filter"
              label="Filter by labels"
              name="filters"
              onChange={this.handleFilters}
              value={filters}
            />
            <Form.Checkbox
              label="Show completed todos"
              name="showCompleted"
              onChange={this.handleShowCompleted}
              checked={showCompleted}
              toggle
            />
          </Form>
        </Popup.Content>
      </Popup>
    );
  }
}

Filters.propTypes = {
  setLabels: PropTypes.func.isRequired,
  setShowCompleted: PropTypes.func.isRequired,
};

Filters.defaultProps = {};

export default connect(mapState, mapDispatch)(Filters);
/* eslint-enable no-shadow */
