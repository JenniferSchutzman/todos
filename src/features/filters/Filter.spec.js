/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Filters from './Filters';

const mockStore = configureStore();
const initialState = { showCompleted: false, labels: '' };
const store = mockStore(initialState);

describe('<Filters />', () => {
  describe('render()', () => {
    test('renders the compnent', () => {
      const wrapper = shallow(<Filters store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
/* eslint-enable react/jsx-filename-extension */
