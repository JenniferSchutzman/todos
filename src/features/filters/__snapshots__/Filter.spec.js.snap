// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> render() renders the compnent 1`] = `
<Filters
  setLabels={[Function]}
  setShowCompleted={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
